apps/cron/src/tasks/processGemsPayout/deployTopReferrerRewardsContract.ts

```diff
@@ -0,0 +1,92 @@
+import { log } from '@charmverse/core/log';
+import { prisma } from '@charmverse/core/prisma-client';
+import { getCurrentSeasonStart, getCurrentSeasonWeekNumber, getLastWeek } from '@packages/dates/utils';
+import { getTopConnectorOfTheDay } from '@packages/scoutgame/topConnector/getTopConnectors';
+import { DateTime } from 'luxon';
+import { parseUnits } from 'viem';
+import { optimismSepolia } from 'viem/chains';
+
+import { createSablierAirdropContract } from './createSablierAirdropContract';
+import { optimismTokenDecimals, optimismTokenAddress } from './deployNewScoutRewardsContract';
+
+const REFERRAL_CHAMPION_REWARD_AMOUNT = parseUnits('25', optimismTokenDecimals).toString();
+
+export async function deployTopReferrerRewardsContract() {
+  const topConnectors: { address: string; date: DateTime }[] = [];
+  const week = getLastWeek();
+  const season = getCurrentSeasonStart(week);
+
+  for (let day = 1; day <= 7; day++) {
+    const date = DateTime.utc().minus({ days: day });
+    const topConnector = await getTopConnectorOfTheDay({ date });
+
+    if (topConnector) {
+      topConnectors.push({ address: topConnector.address, date });
+    }
+  }
+
+  if (topConnectors.length === 0) {
+    log.info('No top connectors found for the week', {
+      week,
+      season
+    });
+    return;
+  }
+
+  if (topConnectors.length === 1) {
+    log.info('Only one top connector found for the week.', {
```

Comment:
if only one person is receiving reward, send something small to NULL address (0x000....)

Reply:
tested and clawback works too